datapipelines/src/main/kotlin/io/customer/datapipelines/plugins/ContextPlugin.kt

```diff
@@ -0,0 +1,33 @@
+package io.customer.datapipelines.plugins
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.segment.analytics.kotlin.core.utilities.putInContextUnderKey
+import io.customer.sdk.data.model.CustomAttributes
+
+/**
+ * Plugin class responsible for updating the context properties in events
+ * tracked by Customer.io SDK.
+ */
+class ContextPlugin(override var analytics: Analytics) : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Before
+
+    internal var deviceToken: String? = null
+        private set
+    internal var attributes: CustomAttributes = emptyMap()
+        private set
+
+    fun updateDeviceProperties(deviceToken: String?, attributes: CustomAttributes) {
+        this.deviceToken = deviceToken
+        this.attributes = attributes
+    }
+
+    override fun execute(event: BaseEvent): BaseEvent {
+        deviceToken?.let { token ->
+            // Device token is expected to be attached to device in context
+            event.putInContextUnderKey("device", "token", token)
```

Comment:
I think this needs to be optional, if `device.token` already exists, should we still override it? this could get problematic in migration or in case device token changes in real time.

Reply:
Good call, I think we should consider this with migration changes. I'll find ticket and update it on Linear too.